cgk_calcium_tools/filehandler.py

```diff
@@ -120,116 +106,200 @@ def __init__(
+                                efocus = [0]
+                            else:
+                                efocus = [get_acquisition_info["Microscope Focus"]]
+                        else:
+                            efocus = [0]
+                            print(
+                                f"Info: Unable to verify Microscope Focus config in: {file}"
+                            )
+                    video.flush()
+                    del video  # usefull for windows
+                    if len(efocus) == 1:
+                        metadata[file]["focus_files"][file] = [file]
+                        metadata[file]["p_rec_paths"].append(file)
+                        metadata[file]["p_recording_labels"].append(
+                            metadata[file]["recording_labels"][0]
+                        )
+                        metadata[file]["efocus"].extend(efocus)
+                    else:
+                        metadata[file]["efocus"].extend(efocus)
+                        pr = Path(file)
+                        efocus_filenames = [
+                            pr.stem + "_" + str(ef) + pr.suffix for ef in efocus
+                        ]
+                        metadata[file]["focus_files"][file] = [
+                            str(Path(metadata[file]["outputsfolders"][0]) / Path(ef))
+                            for ef in efocus_filenames
+                        ]
+                        metadata[file]["p_recording_labels"].extend(
+                            [
+                                metadata[file]["recording_labels"][0] + f"_{ef}"
+                                for ef in efocus
+                            ]
+                        )
+                        metadata[file]["p_rec_paths"].extend(
+                            metadata[file]["focus_files"][file]
+                        )
+                    metadata[file]["p_outputsfolders"].extend(
+                        [metadata[file]["outputsfolders"][0]] * len(efocus)
+                    )
+                for raw_path, intern_data in metadata.items():
+                    json_file = os.path.join(
+                        intern_data["outputsfolders"][0],
+                        os.path.splitext(os.path.basename(raw_path))[0]
+                        + "_metadata.json",
+                    )
+                    # if exist, is remove to write it again with the new data
+                    if os.path.exists(json_file):
+                        os.remove(json_file)
+                    with open(json_file, "w") as j_file:
+                        json.dump(intern_data, j_file)
+                    for key, g in intern_data.items():
+                        if key == "focus_files":
+                            meta[key].update(g)
+                        elif key == "efocus":
+                            meta[key].append(metadata[file][key])
+                        else:
+                            meta[key].extend(metadata[file][key])
```

Comment:
these lines are repeated 3 times. The files can be created under one of the conditions an then for all cases read at the end